torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1001,6 +1002,27 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
   }
 
  private:
+  // Tracks whether a WARN instruction has been executed before. So that we can
+  // ensure each WARN instruction only executes once to mimic Python behavior.
+  struct WarnedNodes {
+   public:
+    bool contains(Node* n) {
```

Comment:
just to be a bit paranoid (not that it matters in this case): there can be a race between two threads between calls to `contains` and `insert` that makes two logs to be logged. You can go by with a single method `shoud_log_once` that both tries to insert and returns a bool whether it was successfully inserted.

Given it's a cheap operation, the simple (not read-write) mutex is fine too